File: solutions/go/0016.go
func threeSumClosest(nums []int, target int) int {
    sort.Ints(nums)
    closest := nums[0] + nums[1] + nums[2]
    for i := 0; i < len(nums) - 2; i++ {
        left := i + 1
        right := len(nums) - 1
        for left < right {
            sum := nums[i] + nums[left] + nums[right]
            if sum == target {
                return sum
            }
            if math.Abs(float64(sum - target)) < math.Abs(float64(closest - target)) {
                closest = sum
            }
            if sum < target {
                left++
            } else {
                right--
            }
        }
    }
    return closest
}

File: solutions/typescript/0016.ts
function threeSumClosest(nums: number[], target: number): number {
    nums.sort((a, b) => a - b);
    let closest = nums[0] + nums[1] + nums[2];
    for (let i = 0; i < nums.length - 2; i++) {
        let left = i + 1;
        let right = nums.length - 1;
        while (left < right) {
            const sum = nums[i] + nums[left] + nums[right];
            if (sum === target) return sum;
            if (Math.abs(sum - target) < Math.abs(closest - target)) {
                closest = sum;
            }
            if (sum < target) {
                left++;
            } else {
                right--;
            }
        }
    }
    return closest;
}